hazelcast/src/main/java/com/hazelcast/config/security/SimpleAuthenticationConfig.java

```diff
@@ -0,0 +1,274 @@
+    private volatile String roleSeparator;
+
+    /**
+     * Adds one user to the configuration.
+     *
+     * @param username username
+     * @param password user's password
+     * @param roles roles assigned to the user
+     * @return this object
+     */
+    public SimpleAuthenticationConfig addUser(@Nonnull String username, @Nonnull String password, String... roles) {
+        userMap.computeIfAbsent(requireNonNull(username), u -> new UserDto(password)).setRoles(roles);
+        return self();
+    }
+
+    /**
+     * Adds one user to the configuration.
+     *
+     * @param username username
+     * @param userDto user's attributes
+     * @return this object
+     */
+    public SimpleAuthenticationConfig addUser(@Nonnull String username, @Nonnull UserDto userDto) {
+        userMap.put(requireNonNull(username, "Username has to be provided"),
+                requireNonNull(userDto, "UserDto object has to be provided"));
+        return self();
+    }
+
+    /**
+     * Allows to use a custom role separator in the configuration. It's important in case when a role name contains the default
+     * separator "," (comma).
+     *
+     * @param roleSeparator new separator
+     * @return this object
+     */
+    public SimpleAuthenticationConfig setRoleSeparator(@Nullable String roleSeparator) {
+        this.roleSeparator = roleSeparator;
+        return self();
+    }
+
+    /**
+     * Returns the custom role separator (when set). If no custom one is used, {@code null} is returned and default separator is
+     * used ("," - comma).
+     *
+     * @return the separator
+     */
+    public String getRoleSeparator() {
+        return roleSeparator;
+    }
+
+    /**
+     * Returns names of users registered in the configuration.
+     *
+     * @return set of usernames
+     */
+    public @Nonnull Set<String> getUsernames() {
+        return new HashSet<>(userMap.keySet());
+    }
+
+    /**
+     * Return the configured password for given username.
+     *
+     * @param username username
+     * @return user's password or null if such user doesn't exist
+     */
+    public String getPassword(@Nonnull String username) {
+        UserDto user = userMap.get(requireNonNull(username, "Username has to be provided"));
+        return user == null ? null : user.password;
+    }
+
+    /**
+     * Returns role names assigned to the given user.
+     *
+     * @param username username
+     * @return configured roles or null if the user doesn't exist
+     */
+    public Set<String> getRoles(@Nonnull String username) {
+        UserDto user = userMap.get(requireNonNull(username, "Username has to be provided"));
+        return user == null ? null : new HashSet<>(user.roles);
+    }
+
+    /**
+     * Replaces the users with ones from the given map.
+     *
+     * @param map map of new users
+     * @return this object
+     */
+    public SimpleAuthenticationConfig setUserMap(@Nullable Map<String, UserDto> map) {
```

Comment:
I'm again not sure about the necessity of this method

Reply:
It's again related to the Spring configuration.